File: AndroidStudioProjects/Workout2021_Step1/app/src/main/java/com/example/workout2021_step1/MainActivity.java
package com.example.workout2021_step1;

import androidx.appcompat.app.AppCompatActivity;

import android.content.Intent;
import android.os.Bundle;
import android.view.View;

public class MainActivity extends AppCompatActivity implements WorkoutListFragment.Listener {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
    }
    public void itemClicked(long id) {
        Intent intent = new Intent(this, DetailActivity.class);
        intent.putExtra(DetailActivity.EXTRA_WORKOUT_ID,(int)id);
        startActivity(intent);
    }

    public void onShowDetails(View v) {
        Intent intent = new Intent(this, DetailActivity.class);
        startActivity(intent);
    }
}

File: AndroidStudioProjects/BMI80/app/src/main/java/com/example/bmi80/MainActivity.kt
package com.example.bmi80

import android.content.Intent
import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.widget.Button
import android.widget.EditText
import android.widget.Toast

class MainActivity : AppCompatActivity() {
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)
        val heightEditText: EditText = findViewById(R.id.heightEditText)
        val weightEditText: EditText = findViewById(R.id.weightEditText)
        val resultButton = findViewById<Button>(R.id.resultButton)
        resultButton.setOnClickListener {
            if(heightEditText.text.isEmpty()||weightEditText.text.isEmpty()){
                Toast.makeText(this, "빈 값이 있습니다.", Toast.LENGTH_SHORT).show()
                return@setOnClickListener
            }
            //isEmpty를 붙여서 유효성 체크하였으니 절대 빈값이 올 수 없다.
            val height: Int = heightEditText.text.toString().toInt()
            val weight: Int = weightEditText.text.toString().toInt()

            val intent = Intent(this, ResultActivity::class.java)
            intent.putExtra("height", height)
            intent.putExtra("weight", weight)

            startActivity(intent)
            finish()
        }
    }
}